src/Microsoft.Diagnostics.Runtime/ClrHeap.cs

```diff
@@ -976,6 +990,43 @@ public override IEnumerable<ulong> EnumerateObjectAddresses()
             }
         }
 
+
+        public override IEnumerable<Tuple<ulong, ulong, ClrType, ulong>> EnumerateObjectDetails()
+        {
+            ClrHeap heap = _segments[0].Heap;
+
+            // optimization for large dumps/apps with 30.000+ types:
+            //    keep track of ClrType given a method table address
+            Dictionary<ulong, ClrType> typeCache = new Dictionary<ulong, ClrType>(32768);
```

Comment:
This is a pretty big allocation that will also always end up on the LOH. The dictionary allocates an int[] buckets list and an Entry[] entries list, both will be on the LOH. It's almost 1MB of data in 64-bit mode.

Reply:
the goal is to avoid triggering garbage collections due to dictionary internal allocations due to array increase: this is important for large memory dumps/address space